python/ray/autoscaler/local/node_provider.py

```diff
@@ -90,12 +111,75 @@ class LocalNodeProvider(NodeProvider):
     """
 
     def __init__(self, provider_config, cluster_name):
+        self.auto_scheduling_mode = False
+        if "server_address" in provider_config:
+            self.auto_scheduling_mode = True
+            self.server_address = provider_config["server_address"]
+            provider_config, cluster_name = self._get_node_configs(
+                provider_config, cluster_name)
+
         NodeProvider.__init__(self, provider_config, cluster_name)
-        self.state = ClusterState("/tmp/cluster-{}.lock".format(cluster_name),
-                                  "/tmp/cluster-{}.state".format(cluster_name),
-                                  provider_config)
+        self.state = ClusterState(
+            "/tmp/cluster-{}.lock".format(cluster_name),
+            "/tmp/cluster-{}.state".format(cluster_name),
+            provider_config,
+        )
+
+    def _get_node_configs(self, provider_config, cluster_name):
+        # Concatenate user_id to make cluster names unique between users.
+        new_cluster_name = cluster_name + "_" + getpass.getuser()
```

Comment:
There's no such thing as a user, let's not add it to the cluster name.

Reply:
It has to. Otherwise, if two separate users use the same cluster name they will interfere with each other.